### DISCUSSION TITLE
Deduplication in Grafana Mimir vs. Thanos

### DISCUSSION THREAD

**Discussion by rosh0450** [grafana/mimir#10223] [General]
I was going through the de-duplication docs for Grafana Mimir https://grafana.com/docs/mimir/latest/configure/configure-high-availability-deduplication/

And then we have de-duplication in Thanos.

The difference which I saw is Grafana Mimir elects a leader Prometheus and offloads metrics from that instance, rejecting metrics from the other instance. While Thanos aggregates the metrics from both instances of Prometheus.

Why was this architecture followed? Aggregating metrics from both instances seems like a good way.

**Comment by pracucci** [comment]
> Why was this architecture followed? Aggregating metrics from both instances seems like a good way.

Aggregating metrics is significantly more expensive than electing a leader.
